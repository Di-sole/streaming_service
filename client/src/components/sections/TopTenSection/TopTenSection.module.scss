.section {
    &__header {
        display: flex;
        flex-direction: row;
        align-items: center;
        margin-bottom: 24px;
    }

    &__title {
        font-size: 24px;
        font-weight: 600;
        color: #ffffff;
    }
}

.topIcon {
    height: 21px;
}
