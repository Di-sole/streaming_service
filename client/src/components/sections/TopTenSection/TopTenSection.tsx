import Image from 'next/image';
import IMovie from '@/models/IMovie';
import topIcon from '../../../../public/icons/posters/top10.svg';
import styles from './TopTenSection.module.scss';
import Slider from '@/components/Slider/Slider';
import PostersList from '@/components/posters/PostersList/PostersList';
import { useTranslation } from 'next-i18next';

interface TopTenSectionProps {
    movies: IMovie[];
}

const TopTenSection = ({ movies }: TopTenSectionProps) => {
    const { t } = useTranslation('mainPage');
    return (
        <div className={styles.section}>
            <div className={styles.section__header}>
                <Image className={styles.topIcon} src={topIcon} alt="топ 10" />
                <h3 className={styles.section__title}>{t('weekTop')}</h3>
            </div>

            <div className={styles.section__content}>
                <Slider itemType="rating" length={movies.length}>
                    <PostersList posterType="rating" movies={movies} />
                </Slider>
            </div>
        </div>
    );
};

export default TopTenSection;
