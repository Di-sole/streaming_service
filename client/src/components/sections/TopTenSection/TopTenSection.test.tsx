import { render, screen } from '@testing-library/react';
import TopTenSection from './TopTenSection';
import { movieTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('TOPTENSECTION SNAPSHOTS TESTS', () => {
    test('TopTenSection should not be changed', () => {
        render(<TopTenSection movies={[movieTest, movieTest]} />);

        const id = screen.getByTestId('topTenSection');
        expect(id).toMatchSnapshot();
    });
});
