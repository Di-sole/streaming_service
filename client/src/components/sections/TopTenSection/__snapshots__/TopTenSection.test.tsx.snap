// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOPTENSECTION SNAPSHOTS TESTS TopTenSection should not be changed 1`] = `
<div
  class="section"
  data-testid="topTenSection"
>
  <div
    class="section__header"
  >
    <img
      alt="топ 10"
      class="topIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <h3
      class="section__title"
    >
      weekTop
    </h3>
  </div>
  <div
    class="section__content"
  >
    <div
      class="jsx-687504307 container container_rating"
      data-testid="slider"
    >
      <div
        class="jsx-687504307 container__contentContainer"
      >
        <div
          class="jsx-687504307 container__content content"
        >
          <div
            class="container ratingItem"
            data-testid="ratingPoster"
          >
            <a
              href="/movies/535341"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="1+1"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
                <div
                  class="container__fadeArea"
                />
              </div>
              <div
                class="container__content"
              >
                <div
                  class="container__fadeArea"
                />
                <div
                  class="container__logoWrapper"
                >
                  <img
                    alt="1+1"
                    class="container__logo"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    src=""
                    style="color: transparent;"
                    width="153"
                  />
                </div>
                <div
                  class="container__numberWrapper"
                >
                  <img
                    alt="undefined"
                    class="container__placeNumber"
                    data-nimg="1"
                    decoding="async"
                    height="66"
                    loading="lazy"
                    src="https://solea-parent.dfs.ivi.ru/picture/bypass/numberundefined.svg"
                    style="color: transparent;"
                    width="48"
                  />
                </div>
              </div>
            </a>
          </div>
          <div
            class="container ratingItem"
            data-testid="ratingPoster"
          >
            <a
              href="/movies/535341"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="1+1"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
                <div
                  class="container__fadeArea"
                />
              </div>
              <div
                class="container__content"
              >
                <div
                  class="container__fadeArea"
                />
                <div
                  class="container__logoWrapper"
                >
                  <img
                    alt="1+1"
                    class="container__logo"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    src=""
                    style="color: transparent;"
                    width="153"
                  />
                </div>
                <div
                  class="container__numberWrapper"
                >
                  <img
                    alt="undefined"
                    class="container__placeNumber"
                    data-nimg="1"
                    decoding="async"
                    height="66"
                    loading="lazy"
                    src="https://solea-parent.dfs.ivi.ru/picture/bypass/numberundefined.svg"
                    style="color: transparent;"
                    width="48"
                  />
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
